src/components/Connections.tsx

```diff
@@ -75,6 +77,7 @@ function Connections() {
   const [xAxis, setXAxis] = useState<string[]>([]);
   const [filteredYAxis, setFilteredYAxis] = useState<HierarchicalItem[]>([]);
   const [filteredXAxis, setFilteredXAxis] = useState<string[]>([]);
+  const [reorderedAxis, setReorderedAxis] = useState<string[]>([]);
```

Comment:
question: I would prefer that we don't introduce a new state variable but rather have the filteredXAxis ordered as desired. Would that be possible?

Reply:
I need the original sorting for the handleClick to see how the data have been reshaped and get the right information when clicking. Unless I move this sorting to an higher point in the flow, which require to re-design the model as well since atm we are missing some info to do the sorting, that would not be possible. For now, considering this apply to a single cluster and so to a subset of the entire data, it's a negligible cost imho and we can keep it here. Thanks!